From 183e15a82c6be6a4f681c668df4436003c3e5364 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 15 Jul 2020 09:00:25 +0300
Subject: [PATCH] [SW-0]: preboot: Fix makefile to include device specific files

Currently, the devices specific files added as part of $(HL_DEV)/Makefile
are getting removed in preboot makefile. This patch adds changes to
fix adding of device specific files along with moving device specific
file to $(HL_DEV)/Makefile.

Change-Id: Ie49674ec5b43fa6aa4e83f99ec2d94ff73778206
---

diff --git a/Makefile b/Makefile
index e9ee0e5..410404d 100644
--- a/Makefile
+++ b/Makefile
@@ -23,16 +23,17 @@
 LDLIBS := -lgcc
 PREBOOT_LDLIBS = $(LDLIBS) -lfdt
 
-include $(HL_DEV)/Makefile
-
 PREBOOT_SRCS_C := $(shell ls common/*.c)
 PREBOOT_SRCS_C += $(shell ls preboot/*.c)
 PREBOOT_SRCS_C := $(filter-out preboot/thermal_v2.c, $(PREBOOT_SRCS_C))
+PPBOOT_SRCS_C := $(shell ls common/*.c)
+PPBOOT_SRCS_C += $(shell ls ppboot/*.c)
+
+include $(HL_DEV)/Makefile
+
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/common.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/$(HL_TARGET).c)
 PREBOOT_APP_S := $(shell ls preboot/crt0_preboot.S)
-PPBOOT_SRCS_C := $(shell ls common/*.c)
-PPBOOT_SRCS_C += $(shell ls ppboot/*.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV)/common.c)
 PPBOOT_SRCS_C += $(shell ls $(HL_DEV)/$(HL_TARGET).c)
 PPBOOT_APP_S := $(shell ls ppboot/crt0_ppboot.S)
@@ -48,8 +49,6 @@
 	LDFLAGS += -Wl,--just-symbols ../arcboot-custom/exports.elf
 else ifeq ($(HL_DEV),goya2_arc_gnu)
 	BINARIES = $(PREBOOT_TARGET)
-	PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/pll.c)
-	PREBOOT_SRCS_C += $(shell ls preboot/thermal_v2.c)
 else
 	BINARIES = $(PPBOOT_TARGET) $(PREBOOT_TARGET) $(PCI_PHY_FW_TARGET)
 endif
diff --git a/goya2/Makefile b/goya2/Makefile
index a448026..d0c80b0 100644
--- a/goya2/Makefile
+++ b/goya2/Makefile
@@ -1,2 +1,2 @@
 FW_SIZE := 0x4400
-PREBOOT_SRCS_C += $(shell ls common/thermal_v2.c)
+PREBOOT_SRCS_C += $(shell ls preboot/thermal_v2.c)
diff --git a/goya2_arc_gnu/Makefile b/goya2_arc_gnu/Makefile
index eacfb1c..c142d30 100644
--- a/goya2_arc_gnu/Makefile
+++ b/goya2_arc_gnu/Makefile
@@ -1 +1,4 @@
 FW_SIZE := 0x8000
+
+PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/pll.c)
+PREBOOT_SRCS_C += $(shell ls preboot/thermal_v2.c)
